src/ucp/core/ucp_context.c

```diff
@@ -574,6 +579,38 @@ const ucp_tl_bitmap_t ucp_tl_bitmap_max = {{UINT64_MAX, UINT64_MAX}};
 const ucp_tl_bitmap_t ucp_tl_bitmap_min = UCS_BITMAP_ZERO;
 
 
+typedef enum ucp_tls_array_search_result {
+    UCP_TLS_ARRAY_SEARCH_RESULT_FOUND     = UCS_BIT(0),
+    UCP_TLS_ARRAY_SEARCH_RESULT_FOUND_AUX = UCS_BIT(1)
+} ucp_tls_array_search_result_t;
+
+
+static int
+ucp_tls_array_check_search_result(uint64_t search_result, uint64_t mask)
+{
+    ucp_tls_array_search_result_t bit;
+
+    ucs_for_each_bit(bit, mask) {
+        if ((search_result & UCS_BIT(bit)) == 0) {
+            return 0;
+        }
+    }
+
+    return 1;
+}
+
+static int ucp_tls_array_found(uint64_t search_result)
+{
+    return ucp_tls_array_check_search_result(search_result,
+                                             UCP_TLS_ARRAY_SEARCH_RESULT_FOUND);
```

Comment:
IMO this can be done much simpler , without ucp_tls_array_check_search_result, since we have only 2 possible bits

Reply:
Removed.